frappe/desk/doctype/dashboard_chart/dashboard_chart.py

```diff
@@ -292,3 +292,38 @@ def check_required_field(self):
 		else:
 			if not self.based_on:
 				frappe.throw(_("Time series based on is required to create a dashboard chart"))
+
+def rename_value_in_dashboard_chart_filter(doc, method, old_docname, new_docname, merge):
+	# called from hooks after a document is renamed
+	import json
+	from frappe.desk.form.linked_with import _get_linked_doctypes
+	linked_doctypes = _get_linked_doctypes(doc.doctype, force_include_all_doctypes=True)
+	doctype_lists = linked_doctypes.keys()
+
+	if not doctype_lists: return
+
+	dashboard_charts = frappe.get_all('Dashboard Chart', fields=['name', 'filters_json', 'document_type'], filters={
+		'document_type': ['in', doctype_lists],
+		'filters_json': ['!=', '{}']
+	})
+
+	for chart in dashboard_charts:
+		chart_filter = frappe.parse_json(chart.filters_json)
+		field_info = linked_doctypes.get(chart.document_type)
+
+		fieldnames = field_info.get('fieldname')
+		doctype_fieldname = field_info.get('doctype_fieldname')
+		child_doctype = field_info.get('child_doctype')
+
+		# considering dashboard chart does not accept dynamic link & child table filters
+		# dashboard_chart.js line 77
+		if doctype_fieldname or child_doctype: continue
+
+		update_required = False
+		for fieldname in fieldnames:
```

Comment:
if we have too many 'fieldnames '(in the case where there is no `document_type`), loop over `chart_filter` and check if it is in 'fieldnames'?